creator-node/src/fileManager.js

```diff
@@ -295,17 +295,35 @@ async function saveFileForMultihashToFS (serviceRegistry, logger, multihash, exp
     }
 
     // verify that the contents of the file match the file's cid
+    let fileData = null
+    let fileSize = null
     try {
-      decisionTree.push({ stage: 'About to verify the file contents for the CID', vals: multihash, time: Date.now() })
+      fileData = (await fs.readFile(expectedStoragePath)).toString().slice(0, 99)
+      fileSize = (await fs.stat(expectedStoragePath)).size
+    } catch (e) {
+      logger.warn('Could not get fs statistics for path', expectedStoragePath)
+    }
+    try {
+      decisionTree.push({ stage: 'About to verify the file contents for the CID', vals: multihash, time: Date.now(), fileData, fileSize })
+
+      // if the file is empty, there's likely something wrong so throw an error
+      // there is one case where an empty file could be valid, check for that CID explicitly
+      if (fileSize !== null && fileSize === 0 && multihash !== 'QmbFMke1KXqnYyBBWxB74N4c5SBnJMVAiMNRcGu6x1AwQH') {
+        throw new Error(`File has no content, content length is 0: ${multihash}`)
+      }
+
       const ipfsHashOnly = await ipfsAdd.ipfsAddNonImages(expectedStoragePath, { onlyHash: true, timeout: 10000 })
       if (multihash !== ipfsHashOnly) {
         decisionTree.push({ stage: `File contents don't match IPFS hash multihash`, vals: ipfsHashOnly, time: Date.now() })
         // delete this file because the next time we run sync and we see it on disk, we'll assume we have it and it's correct
-        await fs.unlink(expectedStoragePath)
```

Comment:
the removeFile in the catch captures both the empty file and hash not matching case

Reply:
nice